src/RAM/Science/components/ClosedLoopControls.tsx

```diff
@@ -0,0 +1,125 @@
+/* eslint-disable prettier/prettier */
+import React, { Component } from 'react';
+import CSS from 'csstype';
+import { rovecomm } from '../../../Core/RoveProtocol/Rovecomm';
+
+const label: CSS.Properties = {
+  marginTop: '-10px',
+  position: 'relative',
+  top: '24px',
+  left: '3px',
+  fontFamily: 'arial',
+  fontSize: '16px',
+  zIndex: 1,
+  color: 'white',
+};
+
+const container: CSS.Properties = {
+  display: 'flex',
+  flexDirection: 'column',
+  fontFamily: 'arial',
+  borderTopWidth: '28px',
+  borderColor: '#990000',
+  borderBottomWidth: '2px',
+  borderStyle: 'solid',
+  padding: '5px'
+};
+
+const row: CSS.Properties = {
+  display: 'flex',
+  flexDirection: 'row',
+  flexGrow: 1,
+  justifyContent: 'space-around',
+  height: '30px',
+  margin: '10px',
+};
+const column: CSS.Properties = {
+  display: 'flex',
+  flexDirection: 'column',
+};
+
+const packetNumList = [0, 1, 2, 3, 4, 5, 6, 7];
+
+interface IProps {
+  style?: CSS.Properties;
+}
+
+interface IState {
+  packetStream: boolean;
+  currentPacket: number;
+}
+
+class ClosedLoopControls extends Component<IProps, IState> {
+  static defaultProps = {
+    style: {},
+  };
+
+  constructor(props:IProps) {
+    super(props);
+    this.state = {
+      packetStream: false,
+      currentPacket: 0,
+    }
+    this.updatePacketStream = this.updatePacketStream.bind(this);
+
+    setInterval(() => this.sendPositionData(), 100);
+  }
+
+  updatePacketStream() {
+    switch(this.state.packetStream) {
+      case false: {
+        this.setState({ packetStream: true });
+        break;
+      }
+      default:
+        this.setState({ packetStream: false });
+        break;
+    }
```

Comment:
You should use the prevState here to invert packetStream instead of using the switch/case. That makes it more readable and helps react on the backend queue state changes. `setGripper()` in Arm.tsx has an example of that.